This patch takes a slightly different approach to solving the issue reported
at https://github.com/NixOS/hydra/issues/200.  This fix allows us to use
Automake's parallel test harness.

--- source/configure.ac.orig	1969-12-31 18:00:01.000000000 -0600
+++ source/configure.ac	2015-04-15 10:58:15.974679278 -0500
@@ -33,7 +33,7 @@
 fi
 ])
 
-NEED_PROG(perl, perl)
+NEED_PROG([PERL], perl)
 
 NEED_PROG([NIX_STORE_PROGRAM], [nix-store])
 
--- source/tests/Makefile.am.orig	1969-12-31 18:00:01.000000000 -0600
+++ source/tests/Makefile.am	2015-04-15 11:00:35.846682904 -0500
@@ -1,19 +1,20 @@
-TESTS_ENVIRONMENT = \
-  BZR_HOME="$(abs_builddir)/data"			\
-  HYDRA_DBI="dbi:SQLite:db.sqlite"			\
-  HYDRA_DATA="$(abs_builddir)/data"			\
-  HYDRA_HOME="$(top_srcdir)/src"			\
-  HYDRA_CONFIG=						\
-  NIX_REMOTE=						\
-  NIX_CONF_DIR="$(abs_builddir)/nix/etc/nix"		\
-  NIX_STATE_DIR="$(abs_builddir)/nix/var/nix"		\
-  NIX_MANIFESTS_DIR="$(abs_builddir)/nix/var/nix/manifests"	\
-  NIX_STORE_DIR="$(abs_builddir)/nix/store"			\
-  NIX_LOG_DIR="$(abs_builddir)/nix/var/log/nix"		\
-  NIX_BUILD_HOOK=					\
-  PERL5LIB="$(srcdir):$(top_srcdir)/src/lib:$$PERL5LIB"	\
-  PATH=$(abs_top_srcdir)/src/script:$(abs_top_srcdir)/src/c:$$PATH \
-  perl -w
+AM_TESTS_ENVIRONMENT = \
+  BZR_HOME="$(abs_builddir)/data"; export BZR_HOME;	\
+  HYDRA_DBI="dbi:SQLite:db.sqlite"; export HYDRA_DBI;	\
+  HYDRA_DATA="$(abs_builddir)/data"; export HYDRA_DATA;	\
+  HYDRA_HOME="$(top_srcdir)/src"; export HYDRA_HOME;	\
+  HYDRA_CONFIG=; export HYDRA_CONFIG;			\
+  NIX_REMOTE=; export NIX_REMOTE;			\
+  NIX_CONF_DIR="$(abs_builddir)/nix/etc/nix"; export NIX_CONF_DIR; \
+  NIX_STATE_DIR="$(abs_builddir)/nix/var/nix"; export NIX_STATE_DIR; \
+  NIX_MANIFESTS_DIR="$(abs_builddir)/nix/var/nix/manifests"; export NIX_MANIFESTS_DIR; \
+  NIX_STORE_DIR="$(abs_builddir)/nix/store"; export NIX_STORE_DIR; \
+  NIX_LOG_DIR="$(abs_builddir)/nix/var/log/nix"; export NIX_LOG_DIR; \
+  NIX_BUILD_HOOK=; export NIX_BUILD_HOOK;		\
+  PERL5LIB="$(srcdir):$(top_srcdir)/src/lib:$$PERL5LIB"; export PERL5LIB; \
+  PATH=$(abs_top_srcdir)/src/script:$(abs_top_srcdir)/src/c:$$PATH; export PATH;
+LOG_COMPILER = $(PERL)
+AM_LOG_FLAGS = -w
 
 EXTRA_DIST = \
   $(wildcard *.pm) \
@@ -33,7 +34,7 @@
 check_SCRIPTS = db.sqlite repos
 
 db.sqlite: $(top_srcdir)/src/sql/hydra-sqlite.sql
-	$(TESTS_ENVIRONMENT) $(top_srcdir)/src/script/hydra-init
+	$(AM_TESTS_ENVIRONMENT) $(TESTS_ENVIRONMENT) $(top_srcdir)/src/script/hydra-init
 
 repos: dirs
 
